compiler/src/iree/compiler/Codegen/Common/TransformExtensions/CommonExtensionsOps.td

```diff
@@ -368,4 +368,40 @@ def ConfigExtractPart :
   }];
 }
 
+def ApplyBufferOptimizationsOp :
+  Op<Transform_Dialect, "iree.apply_buffer_optimizations",
+    [TransformEachOpTrait,
+     TransformOpInterface,
+     DeclareOpInterfaceMethods<MemoryEffectsOpInterface>]> {
+  let description = [{
+    This applies memory optimization on memref. In particular it does store to
+    load forwarding, dead store elimination and dead alloc elimination.
+
+    #### Return modes
+
+    This operation apples a set of memory optimization on the whole region of
+    the operand.
+
+    If the transformation is successful it returns the handle to the
+    same payload as its operand to allow for simpler composition.
+  }];
+
+  let arguments = (ins PDL_Operation:$target);
```

Comment:
see @ftynse's comments on https://reviews.llvm.org/D141860: `Something like TransformHandleTypeInterface is more future-proof and the syntax will need the types to be listed.`

Here you want a `
def Transform_FuncOp : Transform_ConcreteOpType<"func.func">;` (see SCFTransformOps.td for more)